src/EventListener/JsonRequestBodyValidationSubscriber.php

```diff
@@ -162,15 +161,8 @@ private function validateJsonAgainstSchema(Route $route, $decodedJsonRequestBody
      */
     private function throwInvalidRequestException(array $errors): void
     {
-        $errorMessages = array_map(
-            function ($error) {
-                return $error['message'];
-            },
-            $errors
-        );
-
         $exception = new InvalidRequestHttpException('Validation of JSON request body failed.');
-        $exception->setErrors($errorMessages);
+        $exception->setErrors($errors);
```

Comment:
instead of flattening the errors up front, now the complete error structure will be available within the exception